### ISSUE TITLE
One database linked to multiple servers

### ISSUE THREAD

**Issue by Blamo27** [pterodactyl/panel#1694]
**Is your feature request related to a problem? Please describe.**
No related problem ?

**Describe the solution you'd like**
I need to share a database to multiple servers, but the backend is not made for that as
you can see below. We can't add the same database to another server_id, so this feature may need a refactor.
![image](https://user-images.githubusercontent.com/971712/64077278-15f22c80-cccf-11e9-8a5d-a1fba433667e.png)

Just to stay organized, it could be great to add this feature.

**Comment by parkervcp** [comment]
If I am reading this right you want to share on DB across multiple servers?

Pretty sure you can already do that.

**Comment by Blamo27** [comment]
@parkervcp I want to display it on the panel for each servers, i think that I can already use them on different server.

I am looking for a feature allowing us to assign a database for multiple servers. Just to stay organized.

Tell me if my explanation is not clear.

**Comment by DomiiBunn** [comment]
I'd imagine that is the feature there looking for.(Just a mock up) 
![](https://images.bbsnetwork.club/dbmockup.png)

**Comment by Blamo27** [comment]
Yes, the database prefix should be editable and linked to multiple servers. 

Also, a possibility to create "sub users" with less permission to this database ?
So create a users management interface ?

**Comment by schrej** [comment]
You can already share databases with multiple servers, as you already noted. Showing it for multiple servers is a pretty big change, and most people probably won't even use it. This panel is mainly for game server management, and not a tool to manage databases.
As this is probably a pretty rare use case I'd reject this.

**Comment by Blamo27** [comment]
@schrej I think that people will not use database management inside Pterodactyl if this feature is not developed. People that have at least 2 servers generally link their servers with the database, so they will not respond to this GitHub issue.

**Comment by schrej** [comment]
This panel is used by lots of people, no one has requested this feature before you. It didn't even get a lot of likes. So it certainly isn't "everyone".
This feature is mostly directed towards hosting providers that want to provide customers with databases anyways. If you just use the panel for your own servers, and if you have a more complicated setup with shared databases, just mange the databases externally. At least that would be my suggestion.

**Comment by DomiiBunn** [comment]
It would be a nice feature, Maybe 0.8 but i'm not sure if it's worth the effort. Would also pose some issues in the security implementation at least the way i proposed it in the screenshot.

**Comment by dav64** [comment]
Let me bump this issue. I also think it would be a nice feature
Instead of the text input suggested in the screenshot, I rather suggest something like a select box showing all databases bound to the DBMS host

**Comment by TekExplorer** [comment]
this is something i wouldnt mind seeing, for those of us with bungee networks it would make things much easier, imo databases should be by account not by server, though hosts should be able to ofc make them the same

**Comment by matthewpi** [comment]
> this is something i wouldnt mind seeing, for those of us with bungee networks it would make things much easier, imo databases should be by account not by server, though hosts should be able to ofc make them the same

The problem is database hosts are supposed to be linked to a node, meaning if you have multiple servers not on the same node, the database shouldn't be linked.

Another idea would be to have a tab to display all the databases across all servers a user has access to, that way a user can more easily see all their databases in a centralized place instead of having databases "link" to another server.

However, this should probably be done alongside #1980.

**Comment by DaneEveritt** [comment]
I barely have the time to implement features or fix bugs as it is, the more complicated you make this feature request the more likely it is to end up like the other 4 pages of them that never get implemented.

@matthewpi's idea is by far the most straightforward and gets us an MVP out the door without putting tons of time or effort into it.

**Comment by w33ts** [comment]
I'd like to throw my hat in the ring to see some functionality like this, if at all possible. We're using the panel in a large gaming community to manage our game servers and are looking to share the same DB between some bungee servers. For me, it's fine if it's not visible and all but for some of my less technical users it'd be nice. But I also totally understand that it might not be worth the time and effort.

**Comment by Kamryx** [comment]
I'd also love to see this, a more UX friendly way of having a shared database show up in all the respective servers would be sweet

**Comment by roboter5123** [comment]
Bmp as this is really nice if you have multiple mc servers with luckperms. Which is a pretty frequent usecase when hosting multiple minecraft servers. Which is supported due to the bungeecord egg.

**Comment by DaneEveritt** [comment]
In the interest of maintainer sanity and cleaning up an enormous backlog of stale feature requests we have moved to requiring all feature requests be discussed within [Discussions](https://github.com/orgs/pterodactyl/discussions/categories/feature-requests). Please consider re-opening this request there if you believe it is still relevant and worth consideration.

See https://github.com/pterodactyl/panel/issues/5491 for additional context.

Thanks.

<sub>_This comment is in the interest of issue tracker maintenance: it does not imply that maintainers are endorsing or dismissing your request._</sub>
